File: modules/_foundation/src/main/java/com/projecthub/base/workflow/application/service/WorkflowManagementService.java
package com.projecthub.base.workflow.application.service;

import com.projecthub.base.workflow.domain.model.WorkflowDefinition;
import com.projecthub.base.workflow.domain.model.WorkflowContext;
import com.projecthub.base.workflow.domain.repository.WorkflowDefinitionRepository;
import com.projecthub.base.workflow.domain.service.WorkflowEngine;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class WorkflowManagementService {
    private final WorkflowDefinitionRepository workflowRepository;
    private final WorkflowEngine workflowEngine;

    @Transactional(readOnly = true)
    public List<WorkflowDefinition> getAllWorkflows() {
        return workflowRepository.findAll();
    }

    @Transactional(readOnly = true)
    public WorkflowDefinition getWorkflowById(UUID id) {
        return workflowRepository.findById(id)
            .orElseThrow(() -> new IllegalArgumentException("Workflow not found"));
    }

    @Transactional
    public WorkflowDefinition createWorkflow(WorkflowDefinition workflow) {
        // Basic validation
        if (workflow.getStates().isEmpty()) {
            throw new IllegalArgumentException("Workflow must have at least one state");
        }
        if (workflow.getInitialState() == null) {
            throw new IllegalArgumentException("Workflow must have an initial state");
        }

        return workflowRepository.save(workflow);
    }

    @Transactional
    public void executeTransition(UUID workflowId, WorkflowContext context) {
        WorkflowDefinition workflow = getWorkflowById(workflowId);
        workflowEngine.transition(workflow, context);
    }

    @Transactional
    public void deleteWorkflow(UUID id) {
        workflowRepository.deleteById(id);
    }
}

File: src/main/kotlin/com/projecthub/application/workflow/service/WorkflowManagementService.kt
package com.projecthub.core.application.service

import com.projecthub.core.domain.model.workflow.WorkflowDefinition
import com.projecthub.core.domain.model.workflow.WorkflowContext
import com.projecthub.infrastructure.adapter.out.persistence.WorkflowDefinitionRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.util.UUID

@Service
class WorkflowManagementService(
    private val workflowRepository: WorkflowDefinitionRepository,
    private val workflowEngine: WorkflowEngine
) {

    @Transactional(readOnly = true)
    fun getAllWorkflows(): List<WorkflowDefinition> {
        return workflowRepository.findAll()
    }

    @Transactional(readOnly = true)
    fun getWorkflowById(id: UUID): WorkflowDefinition {
        return workflowRepository.findById(id)
            .orElseThrow { IllegalArgumentException("Workflow not found") }
    }

    @Transactional
    fun createWorkflow(workflow: WorkflowDefinition): WorkflowDefinition {
        // Basic validation
        if (workflow.getStates().isEmpty()) {
            throw IllegalArgumentException("Workflow must have at least one state")
        }
        if (workflow.getInitialState() == null) {
            throw IllegalArgumentException("Workflow must have an initial state")
        }

        return workflowRepository.save(workflow)
    }

    @Transactional
    fun executeTransition(workflowId: UUID, context: WorkflowContext) {
        val workflow = getWorkflowById(workflowId)
        workflowEngine.transition(workflow, context)
    }

    @Transactional
    fun deleteWorkflow(id: UUID) {
        workflowRepository.deleteById(id)
    }
}
